lib/Dialect/Seq/Transforms/HWMemSimImpl.cpp

```diff
@@ -755,6 +793,40 @@ void HWMemSimImplPass::runOnOperation() {
     }
   }
 
+  // inline the memory modules created in the previous stage if
+  // inlineMemoryWrapperModules lowering options is enabled
+  if (!wrapperModules.empty() &&
+      LoweringOptions(topModule).inlineMemoryWrapperModules) {
+    SmallVector<InstanceOp> inlinedInstances;
+    auto &symbolTable = getAnalysis<SymbolTable>();
+    for (auto mod :
+         llvm::make_early_inc_range(topModule.getOps<HWModuleOp>())) {
+      for (auto inst : mod.getOps<InstanceOp>()) {
+        auto mem =
+            dyn_cast<HWModuleOp>(symbolTable.lookup(inst.getModuleName()));
+        if (wrapperModules.find(mem) == wrapperModules.end()) {
+          continue;
+        }
+        PrefixingInliner interface(&getContext(), inst.getInstanceName());
+        if (failed(mlir::inlineRegion(interface, &mem.getBody(), inst,
+                                      inst.getOperands(), inst.getResults(),
+                                      std::nullopt, true))) {
+          inst.emitError("failed to inline '")
+              << mod.getModuleName() << "' into instance '"
+              << inst.getInstanceName() << "'";
+          return signalPassFailure();
+        }
+        inlinedInstances.push_back(inst);
+      }
+    }
+    for (auto inst : inlinedInstances) {
+      inst.erase();
+    }
+    for (auto mod : wrapperModules) {
+      mod.erase();
+    }
```

Comment:
Please remove braces: https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements
```suggestion
    for (auto inst : inlinedInstances)
      inst.erase();
    for (auto mod : wrapperModules)
      mod.erase();
```

Reply:
Will do.